### ISSUE TITLE
Warn on Boolean properties beginning in 8.x

### ISSUE THREAD

**Issue by DPUkyle** [gradle/gradle#22218]
A consequence of [GROOVY-10708](https://issues.apache.org/jira/browse/GROOVY-10708) is that Tasks or Extensions having an `is...()` getter returning a `Boolean` are no longer recognized as properties.

One known [example](https://ge.gradle.org/s/2wyscubstyjka/tests/:smoke-test:smokeTest/org.gradle.smoketests.SpringBootPluginSmokeTest/performs%20static%20analysis%20of%20plugin%20org.springframework.boot%20version%202.5.5?focused-execution=1&top-execution=1#L15
) comes from the `spring-boot-gradle-plugin`.  Property `o.s.b.g.t.b.DockerSpec#tlsVerify` is backed by a primitive boolean field, but the accessor returns a `Boolean` [here](https://github.com/spring-projects/spring-boot/blob/v2.5.5/spring-boot-project/spring-boot-tools/spring-boot-gradle-plugin/src/main/java/org/springframework/boot/gradle/tasks/bundling/DockerSpec.java#L68-L76).  This will fail task validation at build time when we upgrade our bundled Groovy to version 4, as shown below:

```
    - Error: Type 'org.springframework.boot.gradle.tasks.bundling.DockerSpec' method 'isTlsVerify()' should not be annotated with: @Input, @Optional.	
      	
      Reason: Input/Output annotations are ignored if they are placed on something else than a getter.	
      	
      Possible solutions:	
        1. Remove the annotations.	
        2. Rename the method.
```

tl;dr

Beginning with Gradle 8.1, we should proactively detect and warn that this pattern will fail in a future version of Gradle where we bundle Groovy 4 - likely Gradle 9.0.

**Comment by big-guy** [comment]
@gradle/bt-kotlin-dsl-maintainers is this something you could look at?
